src/GraphQL/Execution/DocumentExecuter.cs

```diff
@@ -326,4 +328,25 @@ protected virtual ExecutionContext BuildExecutionContext(ExecutionOptions option
         protected virtual IExecutionStrategy SelectExecutionStrategy(ExecutionContext context)
             => _executionStrategySelector.Select(context);
     }
+
+    internal class DocumentExecuter<TSchema> : IDocumentExecuter<TSchema>
+        where TSchema : ISchema
+    {
+        private readonly IDocumentExecuter _documentExecuter;
+        public DocumentExecuter(IDocumentExecuter documentExecuter)
+        {
+            _documentExecuter = documentExecuter ?? throw new ArgumentNullException(nameof(documentExecuter));
+        }
+
+        public Task<ExecutionResult> ExecuteAsync(ExecutionOptions options)
+        {
+            if (options.Schema != null)
+                throw new InvalidOperationException("Schema must be null when calling this typed IDocumentExecuter<> implementation; it will be pulled from the dependency injection provider.");
```

Comment:
```suggestion
                throw new InvalidOperationException("ExecutionOptions.Schema must be null when calling this typed IDocumentExecuter<> implementation; it will be pulled from the dependency injection provider.");
```